pkg/operator/operator.go

```diff
@@ -412,34 +412,36 @@ func (o *Operator) cleanupOldResources(ctx context.Context) error {
 
 	// Cleanup resources without the provided annotation.
 	// Check the collector DaemonSet.
-	var ds appsv1.DaemonSet
-	if err := o.client.Get(ctx, client.ObjectKey{
+	dsKey := client.ObjectKey{
 		Name:      NameCollector,
 		Namespace: o.opts.OperatorNamespace,
-	}, &ds); err != nil && apierrors.IsNotFound(err) {
+	}
+	var ds appsv1.DaemonSet
+	if err := o.client.Get(ctx, dsKey, &ds); apierrors.IsNotFound(err) {
 		o.logger.Error(err, "Getting collector DaemonSet failed")
```

Comment:
I know it's not your problem, but do we really want to double handle the error? It's kind of big anti-pattern (the returned error is likely logged as well or handled in some other way 🤔 )

Reply:
Good point. I suspect that returning the error would be sufficient.

Reply:
Do you want to check it, before merging?

Reply:
Verified that these errors bubble up to the top level of the application, where they are [logged before the program exits](https://github.com/GoogleCloudPlatform/prometheus-engine/blob/main/cmd/operator/main.go#L164). Extra logging messages removed.